### PR TITLE
HG-860 adding alternate link for main pages SEO

### PR DESCRIPTION
**Pull request by lizlux**
We uncovered an SEO issue today (see https://wikia-inc.atlassian.net/browse/HG-860 for more info) and this is one step towards fixing that problem. Basically, we're telling google that there's an alternative version of this page at the supplied URL on mobile. 

Also see https://developers.google.com/webmasters/mobile-sites/mobile-seo/configurations/separate-urls?hl=en for more info.

### CODE REVIEW

File: skins/oasis/modules/templates/Oasis_Index.php [hunk 1]
```diff
@@ -8,6 +8,12 @@
 <?php else : ?>
 	<meta name="viewport" content="width=1200">
 <?php endif ?>
+
+<? if ( Wikia::isMainPage() ): ?>
+	<? global $wgServer ?>
+	<link rel="alternate" media="only screen and (max-width: 640px)" href="<?= $wgServer ?>">
+<? endif; ?>
+
```
Comment (garthwebb) [inline skins/oasis/modules/templates/Oasis_Index.php]: It might be better to have this as part of `$headLinks` which adds stuff like this.  There's even an `SEOTweaks` extension with an `SEOTweaksHooksHelper` class that's already inserting a link:

https://github.com/Wikia/app/blob/HG-860/extensions/wikia/SEOTweaks/SEOTweaksHooksHelper.class.php#L29

Comment (owend) [inline skins/oasis/modules/templates/Oasis_Index.php]: In a template page, you can access $wg->Server without the global reference.

File: extensions/wikia/SEOTweaks/SEOTweaksHooksHelper.class.php [hunk 2]
```diff
@@ -114,11 +117,11 @@ static public function onBeforeParserMakeImageLinkObjOptions( $parser, $title, &
 			$alt = implode( '.', array_slice( explode( '.', $text ), 0, -1 ) ); // lop off text after the ultimate dot (e.g. JPG)
 			$parts[] = "alt={$alt}";
 		}
-		
+
 		return true;
-
```
Comment (lizlux) [inline extensions/wikia/SEOTweaks/SEOTweaksHooksHelper.class.php:119]: My IDE removes unnecessary whitespace, so these can be ignored or viewed with `?w=1`

File: extensions/wikia/SEOTweaks/SEOTweaksHooksHelper.class.php [hunk 3]
```diff
@@ -21,13 +21,16 @@ class SEOTweaksHooksHelper {
...
-		global $wgSEOGoogleSiteVerification, $wgSEOGooglePlusLink;
+		global $wgSEOGoogleSiteVerification, $wgSEOGooglePlusLink, $wgServer;
 		if ( !empty( $wgSEOGoogleSiteVerification ) ) {
 			$out->addMeta( 'google-site-verification', $wgSEOGoogleSiteVerification );
 		}
 		if ( !empty( $wgSEOGooglePlusLink ) ) {
 			$out->addLink( array( 'href' => $wgSEOGooglePlusLink, 'rel' => 'publisher' ) );
 		}
+		if ( Wikia::isMainPage() ) {
+			$out->addLink( [ 'href' => $wgServer, 'rel' => 'alternate', 'media' => 'only screen and (max-width: 640px)' ] );
```
Comment (garthwebb) [inline extensions/wikia/SEOTweaks/SEOTweaksHooksHelper.class.php]: Maybe reorder these so that it outputs in the same order as the other links, e.g. `rel`, `media`, `href`.  Currrently it looks like:

``` html
<link href="http://garth.garth.wikia-dev.com" rel="alternate" media="only screen and (max-width: 640px)" />
<link rel="canonical" href="http://garth.garth.wikia-dev.com/wiki/Garth_Wiki" />
<link rel="alternate" type="application/x-wiki" title="Edit" href="/wiki/Garth_Wiki?action=edit" />
<!-- 7 more links starting w/ rel -->
```

**Comment by kenkouot** [comment]
LGTM

File: extensions/wikia/SEOTweaks/SEOTweaksHooksHelper.class.php (hunk 3 above)
Comment (lizlux) [inline extensions/wikia/SEOTweaks/SEOTweaksHooksHelper.class.php]: Good call, done.

**Comment by garthwebb** [comment]
LGTM
